s2tbx-cache/src/main/java/org/esa/s2tbx/dataio/cache/S2CacheSizeChecking.java

```diff
@@ -36,45 +38,43 @@ public synchronized void setParameters(boolean checkingEnable,double limitSizeCa
      * Method check the S2cache size over time. The method is executed in thread.
      * Each period time, the cache size is checked.
      * The x% of the oldest files should be deleted in case of cache oversize.
-     * @param releasedSpacePurcent The space purcentage of the oldest files should be deleted
-     * @param period The period between two size checking in minute
+     *
+     * @param releasedSpacePercent The space percentage of the oldest files should be deleted
+     * @param period               The period between two size checking in minute
      */
-    public synchronized void launchCacheSizeChecking(double releasedSpacePurcent, int period) {
-        executor = new ThreadExecutor();
-            try {
-                ThreadRunnable runnable = new ThreadRunnable() {
-                    @Override
-                    public void process() throws Exception {
-                        double byteToGigaByte = 0.000000001;
-                        while(!Thread.currentThread().isInterrupted()){
-                            if(checkingEnable){
-                                double currentCacheSize = S2CacheUtils.getCacheSize()*byteToGigaByte;
-                                //compute the limit of cache size should be keep after the checking
-                                double circularLimitSizeCache = limitSizeCache * (1-releasedSpacePurcent);
-                                if(currentCacheSize>limitSizeCache){
-                                    while(currentCacheSize>circularLimitSizeCache)
-                                    {
-                                        File oldestFile = S2CacheUtils.getOldestFolder();
-                                        S2CacheUtils.deleteFile(oldestFile);
-                                        currentCacheSize = S2CacheUtils.getCacheSize() * byteToGigaByte;
-                                    }
-                                }
+    public synchronized void launchCacheSizeChecking(double releasedSpacePercent, int period) {
+        Runnable runnable = () -> {
+                if (checkingEnable) {  // would be better if the executor is stopped when the parameter changes
+                    double currentCacheSize = S2CacheUtils.getCacheSize() * BYTE_TO_GIGA_BYTE;
+                    //compute the limit of cache size should be keep after the checking
+                    double circularLimitSizeCache = limitSizeCache * (1 - releasedSpacePercent);
+                    if (currentCacheSize > limitSizeCache) {
+                        while (currentCacheSize > circularLimitSizeCache) {
+                            File oldestFile = S2CacheUtils.getOldestFolder();
+                            if (oldestFile != null) {
+                                S2CacheUtils.deleteFile(oldestFile);
                             }
-                            Thread.sleep((long)period*20000);
+                            currentCacheSize = S2CacheUtils.getCacheSize() * BYTE_TO_GIGA_BYTE;
```

Comment:
This iterates again over the full folder structure, which may be slow. An alternative would be to do something like:

```java
long folderSize = S2CacheUtils.getFolderSize(oldestFolder);
S2CacheUtils.deleteFolder(oldestFolder); // nb. this might fail
currentCacheSize = Math.max(currentCacheSize / BYTE_TO_GIGA_BYTE - folderSize, 0) * BYTE_TO_GIGA_BYTE`);
```

This avoids iterating over the unaffected files. The the `BYTE_TO_GIGA_BYTE` dance shows that it's probably better to compute `limitSizeCache` in bytes and compare against that, instead of converting `currentCacheSize` to GB.

Reply:
@Florian1209 can you pick up this suggestion?

Reply:
@marpet I pick up both proposals.

Reply:
@marpet Could I squash my commits?